update proposerConfig related sections

tbenr opened the pull request:
# Pull request checklist

Use the following template to make sure your PR fits the Teku documentation standard.

## Describe the change

we changed how `proposerConfig.json` file is interpreted so I described the changes and added and example to clarify the behaviour.

## Issue fixed

fixes #405 

## Impacted parts

<!-- Check the item from the following lists that your PR impacts. You can check multiple boxes. -->

For content changes:

- [x] Documentation content
- [ ] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (for example lint or vale)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [ ] You've [fixed any issues](https://consensys.net/docs/doctools/en/latest/contribute/fix-cicd-errors/) raised by the tests.
- [ ] You've [previewed your changes on Read the Docs](https://consensys.net/docs/doctools/en/latest/preview/old-system/#preview-on-read-the-docs)
  and added a [preview link](#preview).

alexandratran reviewed (approved):
Made some edits for formatting and style. LGTM!

tbenr commented:
@alexandratran I added a general behaviour description and I added another example with a couple of sections in it.
I think it requires a bit of wording\style check!

alexandratran commented on docs/Reference/CLI/CLI-Syntax.md:
```diff
@@ -2475,21 +2475,39 @@ The default is `false`.
+        * `public_key` - (optional in `proposal_config` but forbidden in `default_config`) Public
+          key to be used (instead of the validator's public key) in the validator registration message.
+
+Each attribute value, for a given validator key, will be determined following the priority:
```
```suggestion
Each attribute value, for a given validator key, is determined using the following priority:
```

alexandratran commented on docs/Reference/CLI/CLI-Syntax.md:
```diff
@@ -2475,21 +2475,39 @@ The default is `false`.
 
-* `enabled` - (mandatory when including `builder`) specifies whether to use the [builder endpoint](#builder-endpoint) when proposing blocks.
-* `gas_limit` - (optional) specifies the `gas_limit` for the builder. The default is `30000000`.
+1. specific configuration in `proposer_config`
+1. default configuration `default_config`
+1. default CLI argument (applicable only to `builder.enabled`)
+1. default value (applicable only to `builder.enabled`)
```
```suggestion
1. Specific configuration in `proposer_config`
1. Default configuration in `default_config`
1. Default CLI argument (applicable only to `builder.enabled`)
1. Default value (applicable only to `builder.enabled`)
```

tbenr commented on docs/Reference/CLI/CLI-Syntax.md:
````diff
@@ -2498,20 +2516,182 @@ JSON file that specifies:
+    }
+    ```
+
+    * If `--validators-builder-registration-default-enabled` is NOT specified (or set to false) in CLI
````
@alexandratran the UI isn't that good for those IMO. Don't know if you have other ways to represent these two "* if .." contitions, otherwise we can merge.

alexandratran reviewed (approved):
I ended up moving the proposer configuration information onto its own page: https://pegasys-teku--409.com.readthedocs.build/en/409/HowTo/Configure/Proposer-Configuration/
